bin/mixer_codegen.sh

```diff
@@ -222,7 +222,7 @@ if [ "$optadapter" = true ]; then
   die "config generation failure: $err";
   fi
 
-  go run "$GOPATH/src/istio.io/istio/mixer/tools/mixgen/main.go" adapter -c "$adapteCfdDS" -o "$(dirname "${file}")" "${extraflags}"
+  go run $GOPATH/src/istio.io/istio/mixer/tools/mixgen/main.go adapter -c $adapteCfdDS -o $(dirname "${file}") ${extraflags}
```

Comment:
Ditto.

Reply:
Ex: for this one 
$GOPATH/src/istio.io/istio/bin/mixer_codegen.sh -a mixer/adapter/redisquota/config/config.proto -x "-n redisquota -t quota"

Command with quotes go run "$GOPATH/src/istio.io/istio/mixer/tools/mixgen/main.go adapter" -c "$adapteCfdDS" -o "$(dirname "${file}")" "${extraflags}"

it would expand as  go run "$GOPATH/src/istio.io/istio/mixer/tools/mixgen/main.go adapter" -c "$adapteCfdDS" -o "$(dirname "${file}")" "${extraflags}"
go run "$GOPATH/src/istio.io/istio/mixer/tools/mixgen/main.go adapter" -c 
 "/usr/local/google/home/gargnupur/go/src/istio.io/istio/mixer/adapter/redisquota/config/config.proto_descriptor" -o "/usr/local/google/home/gargnupur/go/src/istio.io/istio/mixer/adapter/redisquota/config" "redisquota -t quota"

and without quotes this expands as
go run $GOPATH/src/istio.io/istio/mixer/tools/mixgen/main.go adapter -c 
 /usr/local/google/home/gargnupur/go/src/istio.io/istio/mixer/adapter/redisquota/config/config.proto_descriptor -o /usr/local/google/home/gargnupur/go/src/istio.io/istio/mixer/adapter/redisquota/config redisquota -t quota

And -t is taken as an option to mixgen without quotes in extraflags and with quotes the whole thing becomes full filename to generate and thus it generate file like "mixer/adapter/redisquota/config/ redisquota -t quota.yaml" which is incorrect

Reply:
Ah I see. So the only problem then is that `extraflags` should be instead expanded as an array.

You'll want to use something like this to split the flags by whitespace robustly:
`IFS=" " read -r -a extraflags_array <<< "$extraflags"`

Then you can use it as:
`"${extraflags_array[@]}"`

The full line would be:
`go run "$GOPATH/src/istio.io/istio/mixer/tools/mixgen/main.go adapter" -c "$adapteCfdDS" -o "$(dirname "${file}")" "${extraflags_array[@]}"`

See these for examples:

- [SC2206](https://github.com/koalaman/shellcheck/wiki/SC2206)
- https://github.com/istio/istio/pull/7694/commits/3811bf42d76d5c17bdaddbffe7a30ccd7bd5175b
- https://github.com/istio/istio/pull/7694/commits/0891171413e04a63ce680acf436493c3ecebeabb

Reply:
Thanks.. that worked!